From 7f09c5d789e72453fafda52ad6f704dba65c8d80 Mon Sep 17 00:00:00 2001
From: Sven Peter <sven@svenpeter.dev>
Date: Sun, 3 Jul 2022 19:19:48 +0200
Subject: [PATCH] fixup! fixup! WIP:/DO-NOT-MERGE: bluetooth

---
 drivers/bluetooth/hci_bcm43xx.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/bluetooth/hci_bcm43xx.c b/drivers/bluetooth/hci_bcm43xx.c
index b66fd2faa6d04e..550171dccecc3b 100644
--- a/drivers/bluetooth/hci_bcm43xx.c
+++ b/drivers/bluetooth/hci_bcm43xx.c
@@ -874,7 +874,7 @@ static const struct firmware *bcm43xx_request_blob(struct bcm43xx_data *bcm43xx,
 	char name[256];
 	int ret;
 
-	snprintf(name, sizeof(name), "brcmbt%s%s-%s-%s.%s", bcm43xx->hw->name,
+	snprintf(name, sizeof(name), "brcm/brcmbt%s%s-%s-%s.%s", bcm43xx->hw->name,
 		 bcm43xx->stepping, bcm43xx->board_type, bcm43xx->vendor,
 		 suffix);
 	dev_info(&bcm43xx->pdev->dev, "Trying to load '%s'", name);
@@ -883,7 +883,7 @@ static const struct firmware *bcm43xx_request_blob(struct bcm43xx_data *bcm43xx,
 	if (!ret)
 		return fw;
 
-	snprintf(name, sizeof(name), "brcmbt%s%s-%s.%s", bcm43xx->hw->name,
+	snprintf(name, sizeof(name), "brcm/brcmbt%s%s-%s.%s", bcm43xx->hw->name,
 		 bcm43xx->stepping, bcm43xx->board_type, suffix);
 	ret = request_firmware(&fw, name, &bcm43xx->pdev->dev);
 	dev_info(&bcm43xx->pdev->dev, "Trying to load '%s'", name);
